### PR TITLE
gateway-api: support ISTIO_MUTUAL  connection to in-mesh gateway

### PR DESCRIPTION
**Pull request by frankbu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/testdata/tls.yaml
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/testdata/tls.yaml:61]: should we just wait for this to land?

**Review by howardjohn — COMMENTED** [review]
LGTM! just one comment

File: pilot/pkg/config/kube/gateway/testdata/tls.yaml
Comment (frankbu) [reply pilot/pkg/config/kube/gateway/testdata/tls.yaml:61]: I could go either way. My thought is that we will wait to use it anywhere (e.g., no docs with this hack), but that it would be nice that the implementation is there as soon as the CRDs are fixed, no code change needed, just a cleanup in the unit test. Anyway, your call if you want to hold this for now.

File: pilot/pkg/config/kube/gateway/conversion.go
Comment (hzxuzhonghu) [inline pilot/pkg/config/kube/gateway/conversion.go:2408]: return

File: pilot/pkg/config/kube/gateway/testdata/tls.yaml
Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/testdata/tls.yaml:61]: Not sure i understand, this cert if used to terminate tls from outside client. Why `ISTIO_MUTUAL` does need this cert?

Comment (frankbu) [reply pilot/pkg/config/kube/gateway/testdata/tls.yaml:61]: This is used for in-mesh clients calling an egress gateway.

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/testdata/tls.yaml:61]: is there a xRoute, and what kind of route is allowed to bind with this gateway

Comment (frankbu) [reply pilot/pkg/config/kube/gateway/testdata/tls.yaml:61]: Istio-specific extensions allow you to bind an xRoute to a `ServiceEntry` to direct traffic to the egress gateway, then bind an xRoute to the gateway to direct traffic to an external `Hostname`. See https://preliminary.istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/#egress-gateway-for-http-traffic

**Comment by frankbu** [comment]
Added DNM to wait for https://github.com/kubernetes-sigs/gateway-api/pull/2721 which looks like it will land soon.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
